src/composables/notifications.ts

```diff
@@ -0,0 +1,155 @@
+import {
+  computed, ref, watch,
+} from '@vue/composition-api';
+import { INotification, INotificationSetting, NotificationStatus } from '../types';
+import {
+  NOTIFICATION_STATUS_CREATED,
+  NOTIFICATION_STATUS_READ,
+  NOTIFICATION_TYPE_WALLET,
+  executeAndSetInterval,
+  watchUntilTruthy,
+} from '../popup/utils';
+import { useGetter, useState } from './vuex';
+import { getAllNotifications, modifyNotifications, normalizeNotification } from '../popup/utils/notifications';
+
+const CHUNK_SIZE = 20;
+
+const superheroNotifications = ref<INotification[]>([]);
+const walletNotifications = ref<INotification[]>([]);
+const pollingInterval = ref<NodeJS.Timer | null>(null);
+
+export function useNotifications(root: any) {
+  const sdk = useState('sdk');
+  const activeNetwork = useGetter('activeNetwork');
+  const account = useGetter('account');
+
+  const canLoadMore = ref(true);
+  const offset = ref(0);
+
+  const notificationSettings = computed<INotificationSetting[]>(
+    () => root.$store.state.notificationSettings,
+  );
+
+  const notificationSettingsCheckedTypes = computed<string[]>(
+    () => notificationSettings.value.filter((s) => s.checked).map((s) => s.type),
+  );
+
+  const filteredNotifications = computed<INotification[]>(
+    () => [...superheroNotifications.value, ...walletNotifications.value]
+      .filter(
+        ({ status, type }) => (
+          status === NOTIFICATION_STATUS_READ
+                  || notificationSettingsCheckedTypes.value.includes(type)
+        ),
+      )
+      .sort((a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime()),
+  );
+
+  const notificationsToShow = computed<INotification[]>(
+    () => filteredNotifications.value.slice(0, offset.value * CHUNK_SIZE),
+  );
+
+  async function getSuperheroNotifications() {
+    const chainNames = useState('chainNames');
+
+    superheroNotifications.value = (
+      await getAllNotifications(sdk.value, activeNetwork.value, account.value.address)
+    ).map(normalizeNotification(chainNames.value));
+  }
+
+  function addWalletNotification(payload: Partial<INotification>) {
+    walletNotifications.value.push({
+      ...payload,
+      type: NOTIFICATION_TYPE_WALLET,
+      status: NOTIFICATION_STATUS_CREATED,
+      createdAt: new Date().toISOString(),
+    });
+  }
+
+  function setWalletNotificationsStatus(createdAt: string, status: NotificationStatus) {
+    const currentNotification = walletNotifications.value.find(
+      (notification) => notification.createdAt === createdAt,
+    );
+    if (currentNotification) {
+      currentNotification.status = status;
+    }
+  }
+
+  function loadMore() {
+    canLoadMore.value = false;
+    offset.value += 1;
+    if (notificationsToShow.value.length < filteredNotifications.value.length) {
+      canLoadMore.value = true;
+    }
+  }
+
+  function handleSetPollingInterval() {
```

Comment:
The word `handle` in my opinion is not necessary. I suggest to rename this and the following methods to:
`startNotificationsPolling`
and
`stopNotificationsPolling`